aws/resource_aws_lex_slot_type_test.go

```diff
@@ -0,0 +1,402 @@
+				),
+			},
+			{
+				ResourceName:            rName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"create_version"},
+			},
+		},
+	})
+}
+
+func TestAccAwsLexSlotType_disappears(t *testing.T) {
+	var v lexmodelbuildingservice.GetSlotTypeOutput
+	resourceName := "aws_lex_slot_type.test"
+	rName := "test_slot_type_" + acctest.RandStringFromCharSet(8, acctest.CharSetAlpha)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsLexSlotTypeNotExists(rName, LexSlotTypeVersionLatest),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsLexSlotTypeConfig_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsLexSlotTypeExists(rName, &v),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsLexSlotType(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsLexSlotTypeExistsWithVersion(slotTypeName, slotTypeVersion string, output *lexmodelbuildingservice.GetSlotTypeOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		var err error
+		conn := testAccProvider.Meta().(*AWSClient).lexmodelconn
+
+		output, err = conn.GetSlotType(&lexmodelbuildingservice.GetSlotTypeInput{
+			Name:    aws.String(slotTypeName),
+			Version: aws.String(slotTypeVersion),
+		})
+		if isAWSErr(err, lexmodelbuildingservice.ErrCodeNotFoundException, "") {
+			return fmt.Errorf("error slot type %s not found, %s", slotTypeName, err)
+		}
+		if err != nil {
+			return fmt.Errorf("error getting slot type %s: %s", slotTypeName, err)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAwsLexSlotTypeExists(slotTypeName string, output *lexmodelbuildingservice.GetSlotTypeOutput) resource.TestCheckFunc {
+	return testAccCheckAwsLexSlotTypeExistsWithVersion(slotTypeName, LexSlotTypeVersionLatest, output)
+}
+
+func testAccCheckAwsLexSlotTypeNotExists(slotTypeName, slotTypeVersion string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).lexmodelconn
+
+		_, err := conn.GetSlotType(&lexmodelbuildingservice.GetSlotTypeInput{
+			Name:    aws.String(slotTypeName),
+			Version: aws.String(slotTypeVersion),
+		})
+		if isAWSErr(err, lexmodelbuildingservice.ErrCodeNotFoundException, "") {
+			return nil
+		}
+		if err != nil {
+			return fmt.Errorf("error getting slot type %s: %s", slotTypeName, err)
+		}
+
+		return fmt.Errorf("error slot type exists %s", slotTypeName)
+	}
+}
+
+func testAccAwsLexSlotTypeConfig_basic(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lex_slot_type" "test" {
+  name = "%s"
+  enumeration_value {
+    synonyms = [
+      "Lirium",
+      "Martagon",
+    ]
+    value = "lilies"
+  }
+  create_version = true
```

Comment:
This configuration should not set `create_version`. In general, the `_basic` configuration should use as many default values as possible